src/LogikUI/Circuit/Wires.cs

```diff
@@ -0,0 +1,126 @@
+                Length = -Length;
+            }
+        }
+    }
+
+
+    class Wires
+    {
+        public Wire[] Powered;
+        public Wire[] UnPowered;
+        public Vector2i[] PoweredConnections;
+        public Vector2i[] UnPoweredConnections;
+
+        public Wires(Wire[] powered, Wire[] unPowered, Vector2i[] poweredConnections, Vector2i[] unPoweredConnections)
+        {
+            PoweredConnections = poweredConnections;
+            UnPoweredConnections = unPoweredConnections;
+            Powered = powered;
+            UnPowered = unPowered;
+        }
+
+        public const double WireWidth = 2;
+        public const double HalfWireWidth = WireWidth / 2;
+
+        public const double ConnectionRadius = 2.5;
+
+        public void Draw(Cairo.Context cr)
+        {
+            WireArray(cr, Powered);
+            cr.SetSourceRGB(0.2, 0.9, 0.2);
+            cr.Fill();
+
+            WireArray(cr, UnPowered);
+            cr.SetSourceRGB(0.1, 0.4, 0.1);
+            cr.Fill();
+
+            foreach (var connection in PoweredConnections)
+            {
+                double x = connection.X * CircuitEditor.DotSpacing;
+                double y = connection.Y * CircuitEditor.DotSpacing;
+                cr.Arc(x, y, ConnectionRadius, 0, Math.PI * 2);
+                cr.ClosePath();
+            }
+            cr.SetSourceRGB(0.2, 0.9, 0.2);
+            cr.Fill();
+
+            foreach (var connection in UnPoweredConnections)
+            {
+                double x = connection.X * CircuitEditor.DotSpacing;
+                double y = connection.Y * CircuitEditor.DotSpacing;
+                cr.Arc(x, y, ConnectionRadius, 0, Math.PI * 2);
+                cr.ClosePath();
+            }
+            cr.SetSourceRGB(0.1, 0.4, 0.1);
+            cr.Fill();
+        }
+
+        public void WireArray(Cairo.Context cr, Wire[] wires)
+        {
+            foreach (var wire in wires)
+            {
+                int vertical = wire.Direction == Direction.Vertical ? 1 : 0;
+                int horizontal = 1 - vertical;
+
+                double x = wire.Pos.X * CircuitEditor.DotSpacing - HalfWireWidth;
+                double y = wire.Pos.Y * CircuitEditor.DotSpacing - HalfWireWidth;
+
+                double length = CircuitEditor.DotSpacing * wire.Length;
+                // If we are drawing a horizontal line the width is length, othervise it's WireWidth.
+                double width = (horizontal * (length + WireWidth)) + (vertical * WireWidth);
+                // The opposite of the above.
+                double height = (vertical * (length + WireWidth)) + (horizontal * WireWidth);
+
+                cr.Rectangle(x, y, width, height);
+
+                // FIXME: Figure out if drawing lines is faster than filling rectangles
+                // To note is that this code needs to be modified to draw the line
+                // to actually cover the grid dots. Rn it just touches them.
```

Comment:
Maybe some of the [linecap](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/stroke-linecap) settings might work to cover the dot?

Reply:
A square linecap would absolutely do the job, question is if it's any faster. I would imagine not as there would be more computation involved, but who knows.

Reply:
After looking at the source for cairo I think rectangles are going to be faster and but mainly easier to implement and get good results from.